include/swift/Frontend/Frontend.h

```diff
@@ -341,11 +352,16 @@ class CompilerInstance {
 
   enum : unsigned { NO_SUCH_BUFFER = ~0U };
   unsigned MainBufferID = NO_SUCH_BUFFER;
+
+  /// Corresponds to PrimaryInput
   unsigned PrimaryBufferID = NO_SUCH_BUFFER;
+  bool generateOutputForTheWholeModule() {
```

Comment:
Naming nitpicks:
- Please make sure predicates are obviously named as predicates; in this case a "should" would probably work best.
- Again, please omit "the" or "a".

However, in this case I think there's a further problem: `performSema` doesn't generate output. `isWholeModule`, `isWholeModuleCompilation`, or the inverse `hasPrimaryFile` would work better.

Reply:
Changed to isWholeModuleCompilation. Someday (not this PR!) we can clean up the code to make the relationships more explicit between which files to type-check, whether main is primary, WMO, etc.